src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.DocumentChangeTracker.cs

```diff
@@ -81,13 +87,25 @@ public void StartTracking(Uri documentUri, SourceText initialText)
                 Contract.ThrowIfTrue(_trackedDocuments.ContainsKey(documentUri), $"didOpen received for {documentUri} which is already open.");
 
                 _trackedDocuments.Add(documentUri, initialText);
+
+                // If we can't find the document in any of the registered workspaces, add it to our loose files workspace.
```

Comment:
I admit I'm not entirely sure why this is happening here versus in RequestContext, but it's possible I don't quite understand the order of what calls what.

Reply:
Doing it here has a couple of useful characteristics which make the code cleaner
1.  We only have to access the lsp misc workspace in the DocumentChangeTracker.  Doing it in the RequestContext creation is possible, but we still have to do it here (to ensure that the document is removed when it is closed when it still exists in the lsp misc workspace).
2.  Document sync notifications do not run concurrently.  So by only modifying the workspace here we don't have to worry about concurrent modifications.  It would also kind of violate the contract that 'NonMutating' requests do not update workspace state.